src/OpenTelemetry/Resources/OtelServiceNameEnvVarDetector.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="OtelServiceNameEnvVarDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Resources
+{
+    internal class OtelServiceNameEnvVarDetector : IResourceDetector
+    {
+        private const string EnvVarKey = "OTEL_SERVICE_NAME";
```

Comment:
Would you mind making this `public const string` (class is already internal) and referencing it in the unit tests so we don't need a duplicate value there? I think this assembly already has internals visible to the tests.

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-dotnet/pull/2209/commits/3268d715d580846c60e3d72eed99e09f8dd073c9